src/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/Zip.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Tools.Zip;
+
+using IOFile        = System.IO.File;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public sealed class Zip : Task
+	{
+		[Required]
+		public  ITaskItem           File            { get; set; }
+
+		public  ITaskItem[]         Entries         { get; set; }
+
+		public  ITaskItem           Prefix          { get; set; }
+
+		public  bool                Overwrite       { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.Low, $"Task {nameof (Zip)}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (File)}: {File.ItemSpec}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Entries)}:");
+			foreach (var p in Entries ?? new ITaskItem [0]) {
+				Log.LogMessage (MessageImportance.Low, $"    {p.ItemSpec}");
+			}
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Overwrite)}: {Overwrite}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Prefix)}: {Prefix?.ItemSpec}");
+
+			if (Overwrite) {
+				IOFile.Delete (File.ItemSpec);
+			}
+
+			var prefix  = Prefix == null ? null : Path.GetFullPath (Prefix.ItemSpec);
+			if (!string.IsNullOrEmpty (prefix) && !prefix.EndsWith (Path.DirectorySeparatorChar.ToString (), StringComparison.OrdinalIgnoreCase)) {
+				prefix  += Path.DirectorySeparatorChar;
+			}
+
+			using (var zip  = ZipArchive.Open (File.ItemSpec, FileMode.OpenOrCreate)) {
+				if (Entries == null)
```

Comment:
Do you need the `Entries` to be `[Required]`? Not sure what this is doing here anyway. Or does the ZipArchive.Open create an empty zip file when it gets disposed, in this case? If yes, then it would be useful to add a comment here.